File: Luna.Pages.Models.View/Models/PageView.cs
namespace Luna.Pages.Models.View.Models;

public class PageView
{
	public Guid Id { get; set; }
	public string Title { get; set; } = null!;
	public string? Description { get; set; }
	public DateTime CreatedAt { get; set; }
	public DateTime UpdatedAt { get; set; }
	public Guid WorkspaceId { get; set; }
	public int LatestVersion { get; set; }
	public Guid OwnerId { get; set; }
	public Guid? ParentId { get; set; }
	public string? Icon { get; set; }
	public string? Cover { get; set; }
	public string? Emoji { get; set; }
	public string Type { get; set; } = null!;
	public string? Path { get; set; }
	public int? Index { get; set; }
	public bool IsTemplate { get; set; }
	public DateTime? ArchivedAt { get; set; }
	public bool Pinned { get; set; }
	public string? CustomSlug { get; set; }
	public object? Properties { get; set; }
}

File: luna-client/src/types/page/pageView.ts
export interface PageView{
	id: string;
	title: string;
	description?: string;
	createdAt: string;
	updatedAt: string;
	workspaceId: string;
	latestVersion: number;
	ownerId: string;
	parentId?: string;
	icon?: string;
	cover?: string;
	emoji?: string;
	type: string;
	path?: string;
	index?: number;
	isTemplate: boolean;
	archivedAt?: string;
	pinned: boolean;
	customSlug?: string;
	properties?: object;
}